### PR TITLE
Set sb_is_modular, sb_is_evergreen appropriately. 

### PR DESCRIPTION
**Pull request by niranjanyardi**
b/294267479
b/294230277

Modular builds are of 2 types:
1. evergreen
2. cobalt toolchain with platform native linker builds

This change consolidates the build system to have mostly 2 flags for modular builds - sb_is_modular, sb_is_evergreen.
1. sb_is_modular : Includes common GN flags and variables related to modular and evergreen GN configurations.
2. sb_is_evergreen: Includes only evergreen-specific GN flags and variables
3. build_with_separate_cobalt_toolchain: Passed as a command line GN arg for cobalt toolchain with platform native linker builds : This flag is used very minimally in the GN code as most of the changes are governed by combinations of sb_is_modular , sb_is_evergreen

This change also cleans up unused GN dependencies 

Change-Id: I8911ec3355ba6d1aa875c96307e7ea0e5a897fc1

### CODE REVIEW

File: starboard/BUILD.gn [hunk 1]
```diff
@@ -57,7 +57,7 @@ group("gn_all") {
     deps += [ "//starboard/benchmark" ]
   }
 
-  if (build_with_separate_cobalt_toolchain && is_cobalt_toolchain) {
+  if (sb_is_modular && !sb_is_evergreen && is_cobalt_toolchain) {
```
[the remark this answers is not in the document]
Comment (niranjanyardi) [reply starboard/BUILD.gn:60]: See b/294436613  on more context as to why im not using it.
TLDR: that variable is not reliable/isn't being set correctly, seems like it works here 
I'll replace it with current_toolchain == cobalt_toolchain .

After the variable/bug is sorted , we can do a wholesale replace of 
current_toolchain == cobalt_toolchain with is_cobalt_toolchain or not

[the remark this answers is not in the document]
Comment (niranjanyardi) [reply starboard/BUILD.gn:60]: this is the only place it's used

[the remark this answers is not in the document]
Comment (niranjanyardi) [reply starboard/BUILD.gn:60]: actually this check is not needed now that i changed the definition of
sb_is_modular = true in evergreen configuration to
sb_is_modular = sb_is_evergreen || build_with_separate_cobalt_toolchain

File: third_party/llvm-project/libcxx/BUILD.gn [hunk 2]
```diff
@@ -119,7 +119,7 @@ static_library("cxx") {
     "//third_party/llvm-project/libcxxabi:cxxabi",
     "//third_party/musl:c",
   ]
-  if (!build_with_separate_cobalt_toolchain) {
+  if (!sb_is_modular || sb_is_evergreen) {
```
[the remark this answers is not in the document]
Comment (niranjanyardi) [reply third_party/llvm-project/libcxx/BUILD.gn:122]: done

File: third_party/llvm-project/libcxxabi/BUILD.gn [hunk 3]
```diff
@@ -90,7 +90,7 @@ static_library("cxxabi") {
   all_dependent_configs = [ ":cxxabi_dependents_config" ]
 
   deps = [ "//third_party/musl:c" ]
-  if (!build_with_separate_cobalt_toolchain) {
+  if (!sb_is_modular || sb_is_evergreen) {
```
[the remark this answers is not in the document]
Comment (niranjanyardi) [reply third_party/llvm-project/libcxxabi/BUILD.gn:93]: done

File: components/update_client/BUILD.gn [hunk 4]
```diff
@@ -247,9 +247,7 @@ target(gtest_target_type, "cobalt_slot_management_test") {
     "//testing/gmock",
     "//testing/gtest",
   ]
-  if (build_with_separate_cobalt_toolchain) {
-    data_deps = [ "//starboard/loader_app($starboard_toolchain)" ]
```
[the remark this answers is not in the document]
Comment (niranjanyardi) [reply components/update_client/BUILD.gn:251]: done

File: third_party/llvm-project/libcxxabi/BUILD.gn (hunk 3 above)
[the remark this answers is not in the document]
Comment (y4vor) [reply third_party/llvm-project/libcxxabi/BUILD.gn:93]: Let's remove the dependency:
"//third_party/llvm-project/libunwind:unwind_evergreen"

**Review by y4vor — APPROVED** [review]

File: cobalt/BUILD.gn [hunk 5]
```diff
@@ -81,8 +81,5 @@ group("default") {
         "//components/update_client",
       ]
     }
-    if (!build_with_separate_cobalt_toolchain) {
-      deps += [ "//third_party/llvm-project/libunwind:unwind_evergreen" ]
```
Comment (andrewsavage1) [inline cobalt/BUILD.gn:85]: Where does this dep now come from?

File: starboard/BUILD.gn (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrewsavage1) [reply starboard/BUILD.gn]: fyi is_cobalt_toolchain will always work in BUILD.gn files and GNI files imported by BUILD.gn files

File: cobalt/BUILD.gn (hunk 5 above)
Comment (andrewsavage1) [reply cobalt/BUILD.gn:85]: I guess it comes from starboard—I'm surprised that doesn't cause dependency check issues (is it in public_deps?), but this seems fine

File: components/update_client/BUILD.gn [hunk 6]
```diff
@@ -246,10 +246,6 @@ target(gtest_target_type, "cobalt_slot_management_test") {
...
     "//testing/gtest",
+    "//starboard/loader_app"
   ]
-  if (build_with_separate_cobalt_toolchain) {
-    data_deps = [ "//starboard/loader_app($starboard_toolchain)" ]
-  } else {
-    deps += [ "//starboard/loader_app" ]
-  }
-}
+}
```
[the remark this answers is not in the document]
Comment (niranjanyardi) [reply components/update_client/BUILD.gn]: done

**Review by andrewsavage1 — APPROVED** [review]
